### ISSUE TITLE
0.6: parser freezes in `using ApproxFun`

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaLang/julia#20450]
I'm trying to update ApproxFun to support Julia 0.6, but it's freezing on line https://github.com/JuliaApproximation/ApproxFun.jl/blob/support-0.6/src/Multivariate/LowRankFun.jl#L28
```julia
LowRankFun{S,M,T,V}(A::Vector{Fun{S,T}},B::Vector{Fun{M,V}}) =
 LowRankFun(convert(Vector{Fun{S,promote_type(T,V)}},A),convert(Vector{Fun{M,promote_type(T,V)}},B),space(first(A))⊗space(first(B)))
```
You can reproduce it as follows:
```julia
Pkg.add("ApproxFun")
Pkh.checkout("FastTransforms")
Pkg.checkout("BandedMatrices")
Pkg.checkout("FastGaussQuadrature","support-0.6")
Pkg.checkout("ApproxFun","support-0.6")
Pkg.test("ApproxFun")
```
I've left debug statements, which read
```julia
Domains
Multivariate
VectorFun
TensorSpace
LowRankFun
x
x
x
```
and then freezes (the "x" printouts are in LowRankFun.jl).

**Comment by dlfivefifty** [comment]
```julia
julia> versioninfo()
Julia Version 0.6.0-dev.2584
Commit 509c775 (2017-02-04 00:17 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin16.4.0)
  CPU: Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz
  WORD_SIZE: 64
  BLAS: libgfortblas
  LAPACK: liblapack
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, ivybridge)
```

**Comment by JeffBezanson** [comment]
I put in a workaround. See if this helps.

**Comment by dlfivefifty** [comment]
Thanks!  It looks like that issue is fixed

**Comment by tkelman** [comment]
Should this be closed then, or left open until the skipped cases also work?

**Comment by JeffBezanson** [comment]
Also probably fixed by #20698.
